sklearn/tree/_splitter.pyx

```diff
@@ -406,7 +406,8 @@ cdef class BestSplitter(BaseDenseSplitter):
 
                             current_proxy_improvement = self.criterion.proxy_impurity_improvement()
 
-                            if current_proxy_improvement > best_proxy_improvement:
+                            if current_proxy_improvement > best_proxy_improvement or \
+                                    (current_proxy_improvement == best_proxy_improvement and current.feature < best.feature):
```

Comment:
If we want to pursue with those changes, we should probably make the motive explicit.
```suggestion
                            if (current_proxy_improvement > best_proxy_improvement or
                                # Breaking ties on the feature with lowest index
                                (current_proxy_improvement == best_proxy_improvement and
                                 current.feature < best.feature)):
```